### ISSUE TITLE
terraform show from plan should show module details

### ISSUE THREAD

**Issue by mattlawnz** [hashicorp/terraform#21949]
### Current Terraform Version

```
Terraform v0.12.3
+ provider.aws v2.17.0
```

### Use-cases
We are creating plans from our terraform files, which we are wanting to put into a git repo as human readable text. In order to get this in a human readable format, we are doing a "terraform show plan.tfplan"
Where we have terraform modules involved, the terraform show command from the precreated t.tfplan only shows the module summary lines instead of any details

e.g.
```
terraform show t.tfplan
  + module.db.module.db_instance.aws_db_instance.this[0]

  ? module.db.module.db_parameter_group.aws_db_parameter_group.this[0]

  ? module.db.module.db_subnet_group.aws_db_subnet_group.this[0]
```

### Attempted Solutions
I think I found a way around, but its a bit clumsy at present

```terraform12 plan -no-color  > plan.txt```

and then reading and importing that into git

### Proposal

Would like options to be able to show module details.

**Comment by mildwonkey** [comment]
Hi @mattlawnz, thank you for opening this enhancement request !

Can you please share some more information about what exact module details you are looking for, or what you'd like the output to look like? 

You might be able to get the information you want from the output of [`terraform show -json plan.tfplan`](https://www.terraform.io/docs/internals/json-format.html), which includes the plan, current state, and configuration.

**Comment by mattlawnz** [comment]
Hi

Thanks for the response, you're right, I can get the output using that.
I was looking for a more human readable format that contains the details for the modules.
We would like to put the output into a git pull request in a human readable format.
If the plan doesn't contain any modules, we can create this human readable plan using our current method. If it does contain modules, it gets a bit messy

**Comment by apparentlymart** [comment]
This was a bug, and was fixed in #23292. The fix should be included in the next minor release. Thanks for reporting this, @mattlawnz!
